### PR TITLE
Plugin/mlflow custom flavor - fixed small bug

### PR DESCRIPTION
**Pull request by jernejfrank**
Thanks for adding the custom flavor option to mlflow. I tested it and only the flavor specific kwargs somehow got mixed up and tried to be called somewhere else which lead to a fatal error.

## Changes
- renamed class attribute to avoid conflicting calls

## How I tested this
- added unit test that doubles as minimal example for reproducing the error

## Notes
- only tested the DataSaver (due to internal legacy structure, DataLoader is not utilised/hard to implement)

## Checklist

- [x] PR has an informative and human-readable title (this will be pulled into the release notes)
- [x] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Comment by skrawcz** [comment]
Thanks @jernejfrank ! For the pre-commit failure you just need to do something like:

`pip install pre-commit`
`pre-commit install`
`pre-commit run --all-files`
And then check that change in  and you should be good to go. Thanks for the fix!

**Comment by skrawcz** [comment]
I'll plan to push this in the release on Monday. If you don't get to the `pre-commit` before I'll do it. Thanks!

File: examples/hamilton_ui/data_set_v1.parquet
Comment (skrawcz) [inline examples/hamilton_ui/data_set_v1.parquet:1]: delete this please

File: examples/hamilton_ui/lr_model.pkl
Comment (skrawcz) [inline examples/hamilton_ui/lr_model.pkl:1]: delete this change please

File: examples/hamilton_ui/svm_model.pkl
Comment (skrawcz) [inline examples/hamilton_ui/svm_model.pkl:1]: delete this change please

**Review by skrawcz — CHANGES_REQUESTED** [review]
looks good. just a few things:

1. pre-commit.
2. you added some files in by accident I assume from the hamilton_ui example.

File: examples/hamilton_ui/data_set_v1.parquet
Comment (jernejfrank) [reply examples/hamilton_ui/data_set_v1.parquet:1]: missed this got generated, sorry.

File: examples/hamilton_ui/lr_model.pkl
Comment (jernejfrank) [reply examples/hamilton_ui/lr_model.pkl:1]: same here.

File: examples/hamilton_ui/svm_model.pkl
Comment (jernejfrank) [reply examples/hamilton_ui/svm_model.pkl:1]: and here.

**Review by skrawcz — APPROVED** [review]

**Comment by skrawcz** [comment]
Thanks @jernejfrank !

**Pull request merged**